portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,36 +19,55 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import java.util.ArrayList;
 import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 import com.andrewrs.sps.data.ListRecord;
-import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     private ArrayList<ListRecord> records;
-  public void init()
-  {
-    records = new ArrayList<ListRecord>();    
-  }
+  public void init(){}
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException 
   {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    records = new ArrayList<ListRecord>(); 
```

Comment:
Because you're initializing `records` in the `doGet` method, there's no benefit to declaring `private ArrayList<ListRecord> records` at the top of this class (because the variable can't be used in any other methods).

I'd recommend changing this to `List<ListRecord> records = ...`

Reply:
Changed that, thanks for pointing it out!